tests/garage/tf/regressors/test_bernoulli_mlp_regressor_with_model.py

```diff
@@ -0,0 +1,233 @@
+import pickle
+from unittest import mock
+
+from nose2.tools.params import params
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.optimizers import ConjugateGradientOptimizer, LbfgsOptimizer
+from garage.tf.regressors import BernoulliMLPRegressorWithModel
+from tests.fixtures import TfGraphTestCase
+
+
+def get_labels(input_shape, xs):
+    label = [0, 0]
+    if input_shape == (1, ):
+        ys = 0 if np.sin(xs) <= 0 else 1
+        label[ys] = 1
+
+    elif input_shape == (2, ):
+        ys = int(np.round(xs[0])) ^ int(np.round(xs[1]))
+        label[ys] = 1
+
+    return label
+
+
+def get_train_data(input_shape):
+    if input_shape == (1, ):
+        data = np.linspace(-np.pi, np.pi, 1000)
+        obs = [{
+            'observations': [[x]],
+            'returns': [get_labels(input_shape, x)]
+        } for x in data]
+
+    elif input_shape == (2, ):
+        data = [np.random.rand(2) for _ in range(1000)]
```

Comment:
please don't use random data in tests. it makes it difficult to make them deterministic, and usually take longer than necessary.

Reply:
sure. will look at alternative ways to test.